martin/build.rs

```diff
@@ -1,13 +1,115 @@
-fn main() -> std::io::Result<()> {
-    #[cfg(feature = "webui")]
-    {
-        static_files::NpmBuild::new("martin-ui")
-            .install()?
-            .run("build")?
-            .target("martin-ui/dist")
-            .change_detection()
-            .to_resource_dir()
-            .build()?;
+#[cfg(feature = "webui")]
+use std::fs;
+#[cfg(feature = "webui")]
+use std::path::Path;
+
+#[cfg(feature = "webui")]
+fn copy_file_tree(src: &Path, dst: &Path, exclude_dirs: &[&str]) {
+    assert!(
+        src.is_dir(),
+        "source for the copy operation is not an existing directory"
+    );
+    let _ = fs::remove_dir_all(dst); // ignore if dir does not exist
+    fs::create_dir_all(dst).unwrap_or_else(|e| {
+        panic!(
+            "failed to create destination directory {}: {e}",
+            dst.display()
+        )
+    });
+    let excludes = exclude_dirs.iter().map(|v| src.join(v)).collect::<Vec<_>>();
+
+    let mut it = walkdir::WalkDir::new(src).into_iter();
+    loop {
+        let entry = match it.next() {
+            None => break,
+            Some(Err(err)) => panic!("failed to read directory entry: {err}"),
+            Some(Ok(entry)) => entry,
+        };
```

Comment:
more of a nit: I would find while-let to be more readable than loop-break

```suggestion
    while let Some(entry) = it.next() {
        let entry = match entry {
            Ok(entry) => entry,
            Err(err) => panic!("failed to read directory entry: {err}"),
        };
```

Reply:
good catch! I forgot about `while let` (copied `walkdir` example for skipping dirs). Going further, I think we should use `if let Err(...) = .. {panic}` too